### DISCUSSION TITLE
iOS - Images not showing following upgrade to v16.1.0

### DISCUSSION THREAD

**Discussion by yousrasd** [invertase/react-native-firebase#6726] [Q&A] [answered]
Hello,

In our app, we use firebase messaging to send push notifications. Our app has a notification service extension (https://rnfirebase.io/messaging/ios-notification-images) so that images sent along a push notification appear. 
Previously, our app was using firebase/messaging v 14.10.1. Images in push notifications appeared.
But following the upgrade of our app to firebase/messaging v 16.1.0, images no more show up when a push notification is received on an iphone. This **works fine** on **Android** device. So the issue is **only on iOS**.
Ive been looking at ways to fix this, but so far I do not see what is wrong. I looked at the change log of react-native-firebase (diff between v14.10.1 and 16.1.0) but i didnt spot any major change related to images.

Is anyone running into the same issue or has any clue/idea on how to fix this?

**Comment by mikehardy** [comment, marked as answer]
:thinking: hmmm

This sounds like an issue, but is missing all the details that could lead to anyone helping - https://stackoverflow.com/help/how-to-ask / https://stackoverflow.com/help/minimal-reproducible-example

The main difference between v14 and v16+ is the move to firebase-ios-sdk versions that require `use_frameworks! :linkage => :static` - I'm not sure if that could affect this or not. It would be interesting to know exactly where the problem happens, with regard to versions. What I mean specifically is a bisect of the versions which (if I remember my math correctly...) will provably result in an answer in 7 or fewer tests, and typically more like 3-4. Our https://github.com/invertase/react-native-firebase/blob/main/CHANGELOG.md contains details of every change for every release, so knowing which version breaks could help isolate why it breaks for you

**Comment by yousrasd** [comment]
Thank you @mikehardy for your reply

The issue is actually the use_frameworks!. When we upgraded to the latest version, we did not add it. 
When I tried adding it, I run into bunch of linking issues with GoogleMaps, so had to upgrade to RN 0.70 first. And now, I retried upgrading to the latest react-native-messaging (I declared the use_frameworks!) and it works!! We are getting push notifications with images
